### ISSUE TITLE
Support `--cascade` option on deletion.

### ISSUE THREAD

**Issue by fullykubed** [alekc/terraform-provider-kubectl#145]
When Kubernetes removes certain controllers like StatefulSets, you have the ability to control whether it also deletes the underlying resources such as the Pods / PVCs ([reference](https://kubernetes.io/docs/tasks/run-application/delete-stateful-set/#deleting-a-statefulset)).

It would be nice to be able to configure this setting to be able to retain some resources after deleting the main controllers.

**Comment by fullykubed** [comment]
I agree. However, the purpose of this is actually to aid in making updates to controllers.

Many Kubernetes controllers such as StatefulSets have immutable fields. For example, in StatefulSets, `.spec.volumeClaimTemplates` is immutable. 

As a result, the only way to update this field is to delete the StatefulSet and create a new one.  The only way to do that without deleting all of the Pods and PVCs is via `--cascade=orphan`. 

This impacts many controllers, both ones built-in to Kubernetes and third-party controllers as well.

Having the ability to specify `--cascade` would easy a burden that currently requires intervention outside of IAC.

Now in a perfect world, we would also be able to apply this only on "recreate" and not just "delete," but my understanding is that is not very straightforward?

**Comment by stevehipwell** [comment]
@fullykubed now that #165 has been merged you should be able to easily add the orphan cascade option to the `delete_cascade` input.

**Comment by fullykubed** [comment]
@stevehipwell Thanks for the heads up.

@alekc Would it be fine to add orphan to the list of available options for `delete_cascade`?

**Comment by alekc** [comment]
Implemented in `kubectl_manifest`. The `delete_cascade` attribute accepts `Background` or `Foreground`:

```hcl
resource "kubectl_manifest" "stateful" {
  delete_cascade = "Foreground"  # or "Background"
  yaml_body      = "..."
}
```

When unset, the provider defaults to `Foreground` if `wait = true`, otherwise `Background` (matches the historic implicit behaviour). Validation rejects any other value. Docs: \`docs/resources/kubectl_manifest.md\`.

Closing.
